src/platforms/common/blackpill-f4/blackpill-f4.c

```diff
@@ -173,24 +174,98 @@ void platform_target_clk_output_enable(bool enable)
 
 bool platform_spi_init(const spi_bus_e bus)
 {
-	(void)bus;
-	return false;
+	uint32_t controller = 0;
+	if (bus == SPI_BUS_INTERNAL) {
+		/* Set up onboard flash SPI GPIOs: PA5/6/7 as SPI1 in AF5, PA4 as nCS output push-pull */
+		gpio_mode_setup(OB_SPI_PORT, GPIO_MODE_AF, GPIO_PUPD_NONE, OB_SPI_SCLK | OB_SPI_MISO | OB_SPI_MOSI);
+		gpio_mode_setup(OB_SPI_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, OB_SPI_CS);
+		gpio_set_output_options(
+			OB_SPI_PORT, GPIO_OTYPE_PP, GPIO_OSPEED_50MHZ, OB_SPI_SCLK | OB_SPI_MISO | OB_SPI_MOSI | OB_SPI_CS);
+		gpio_set_af(OB_SPI_PORT, GPIO_AF5, OB_SPI_SCLK | OB_SPI_MISO | OB_SPI_MOSI);
+		/* Deselect the targeted peripheral chip */
+		gpio_set(OB_SPI_PORT, OB_SPI_CS);
+	} else if (bus == SPI_BUS_EXTERNAL) {
+		/* Set up external SPI GPIOs: PB13/14/15 as SPI2 in AF5, PB12 as nCS output push-pull */
+		gpio_mode_setup(EXT_SPI_PORT, GPIO_MODE_AF, GPIO_PUPD_NONE, EXT_SPI_SCLK | EXT_SPI_MISO | EXT_SPI_MOSI);
+		gpio_mode_setup(EXT_SPI_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, EXT_SPI_CS);
+		gpio_set_output_options(
+			EXT_SPI_PORT, GPIO_OTYPE_PP, GPIO_OSPEED_50MHZ, EXT_SPI_SCLK | EXT_SPI_MISO | EXT_SPI_MOSI | EXT_SPI_CS);
+		gpio_set_af(EXT_SPI_PORT, GPIO_AF5, EXT_SPI_SCLK | EXT_SPI_MISO | EXT_SPI_MOSI);
+		/* Deselect the targeted peripheral chip */
+		gpio_set(EXT_SPI_PORT, EXT_SPI_CS);
+	} else
+		return false;
+
+	if (bus == SPI_BUS_INTERNAL) {
+		rcc_periph_clock_enable(RCC_SPI1);
+		rcc_periph_reset_pulse(RST_SPI1);
+		controller = OB_SPI;
+	} else if (bus == SPI_BUS_EXTERNAL) {
+		rcc_periph_clock_enable(RCC_SPI2);
+		rcc_periph_reset_pulse(RST_SPI2);
+		controller = EXT_SPI;
+	}
```

Comment:
Rather than this being a separate if block, especially with where the `controller` variable is defined, it would make more sense for this to be integrated with the above block with a line of whitespace between the GPIO setup and the RCC pokes. That would make this flow better.

Reply:
After modifying gpio setup to conform to code style like in native, I agree, moving these inside works better.

Reply:
Propagated RCC upwards, keeping the `controller` variable. I could duplicate the SPI init code, too, setting different divisors (because SPI1 & SPI2 are on APB2 & APB1 and one is clocked twice as fast than the other), but for that I think a better scheme is needed which allows e.g. instructing BMP from `bmpflash` on divisors in a `monitor frequency` fashion.